src/k2/module/k23si/Module.cpp

```diff
@@ -1237,76 +1232,81 @@ K23SIPartitionModule::handleTxnFinalize(dto::K23SITxnFinalizeRequest&& request)
     VersionSet& versions = IndexerIt->second;
     bool found = false;
     bool isWI = false;
-    bool isAborted = false;
 
     // Check for matching WI first
     if (versions.WI.has_value() && versions.WI->txnId == txnId) {
         found = true;
         isWI = true;
-        isAborted = versions.WI->status == dto::WriteIntent::Status::Aborted;
     }
     // Check for a matching committed record if needed
     if (!found) {
         for (const CommittedRecord& record : versions.committed) {
             // timestamps are unique, so they can identify a txn match
             if (record.timestamp.compareCertain(txnId.mtr.timestamp) == 0) {
```

Comment:
I think it'd be slightly more efficient(fewer branches) to compare once and break, something like
``` c++
int comp = record.timetamp.compareCertain(txnId.mtr.timestamp);
found = !comp;
if (comp <= 0) break;
```